### PR TITLE
fix(oauth): Show oauth apps to non admin users as well

### PR DESCRIPTION
**Pull request by realaashil**
Shows personal apps (created by them) to non admin users

## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/nodejs/apps/src/modules/user_management/services/user-admin.service.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  userId: string,
+  orgId: string,
+): Promise<boolean> => {
+  const groups = await UserGroups.find({
+    orgId,
+    users: { $in: [userId] },
+    isDeleted: false,
+  }).select('type');
+
+  return groups.some((userGroup: any) => userGroup.type === 'admin');
```
[the remark this answers is not in the document]
Comment (realaashil) [reply backend/nodejs/apps/src/modules/user_management/services/user-admin.service.ts]: fixed

File: backend/nodejs/apps/src/modules/oauth_provider/services/oauth.app.service.ts [hunk 2]
```diff
@@ -253,26 +281,29 @@ export class OAuthAppService {
...
+      ...this.buildAppFilter(orgId, userId, isAdmin),
     })
 
     if (!app) {
       throw new NotFoundError('OAuth app not found')
     }
 
     app.isDeleted = true
-    app.deletedBy = new Types.ObjectId(deletedBy)
     app.status = OAuthAppStatus.REVOKED
```
[the remark this answers is not in the document]
Comment (realaashil) [reply backend/nodejs/apps/src/modules/oauth_provider/services/oauth.app.service.ts:293]: fixed

File: backend/nodejs/apps/src/modules/oauth_provider/schema/oauth.app.schema.ts [hunk 3]
```diff
@@ -108,7 +107,6 @@ const OAuthAppSchema = new Schema<IOAuthApp>(
     accessTokenLifetime: { type: Number, default: 3600 },
     refreshTokenLifetime: { type: Number, default: 2592000 },
     isDeleted: { type: Boolean, default: false },
```
[the remark this answers is not in the document]
Comment (realaashil) [reply backend/nodejs/apps/src/modules/oauth_provider/schema/oauth.app.schema.ts:111]: fixed

File: backend/nodejs/apps/src/modules/oauth_provider/schema/oauth.app.schema.ts [hunk 4]
```diff
@@ -33,7 +33,6 @@ export interface IOAuthApp extends Document {
   accessTokenLifetime: number
   refreshTokenLifetime: number
   isDeleted: boolean
```
[the remark this answers is not in the document]
Comment (realaashil) [reply backend/nodejs/apps/src/modules/oauth_provider/schema/oauth.app.schema.ts:35]: fixed

File: backend/nodejs/apps/src/modules/oauth_provider/routes/oauth.clients.routes.ts [hunk 5]
```diff
@@ -26,8 +25,6 @@ export function createOAuthClientsRouter(container: Container): Router {
 
   // All routes require authentication
   router.use(authMiddleware.authenticate.bind(authMiddleware))
-  // All routes require admin auth
-  router.use(userAdminCheck)
   // All routes are rate limited
```
[the remark this answers is not in the document]
Comment (realaashil) [reply backend/nodejs/apps/src/modules/oauth_provider/routes/oauth.clients.routes.ts:28]: fixed

**Comment by realaashil** [comment]
@claude

**Comment by realaashil** [comment]
@claude

File: frontend-new/app/(main)/workspace/developer-settings/oauth2/components/manage-oauth-application-panel.tsx [hunk 6]
```diff
@@ -583,6 +583,7 @@ export function ManageOAuthApplicationPanel({
         homepageUrl: optionalUrlOrEmpty(homepageUrl),
         privacyPolicyUrl: optionalUrlOrEmpty(privacyPolicyUrl),
         termsOfServiceUrl: optionalUrlOrEmpty(termsOfServiceUrl),
+        logoUrl: null,
```
[the remark this answers is not in the document]
Comment (realaashil) [reply frontend-new/app/(main)/workspace/developer-settings/oauth2/components/manage-oauth-application-panel.tsx]: fixed

**Review by rish664 — APPROVED** [review]
